src/main/java/com/leaf/billing/utils/PropertyBuilder.java

```diff
@@ -15,18 +23,57 @@ private PropertyBuilder() {
     public static final String TARGET_BASE_PATH = "targetBasePath";
     public static final String SOURCE_DELETE_FILES = "sourceDeleteFiles";
 
-    public static JobProperties build(String... args) {
+    public static JobProperties build(String... args) throws IOException, URISyntaxException, InvalidPathException {
         Map<String, String> sparkProperties = Arrays.stream(args)
                 .filter(arg -> arg.contains("="))
                 .map(arg -> arg.split("=", 2))
                 .filter(keyValue -> keyValue.length == 2)
                 .collect(Collectors.toMap(keyValue -> keyValue[0], keyValue -> keyValue[1]));
+
+        String normalizedPathSource = Optional.ofNullable(sparkProperties.get(SOURCE_BASE_PATH)).map(PropertyBuilder::normalizePath).orElseThrow(InvalidPathException::new);
+        String normalizedPathTarget = Optional.ofNullable(sparkProperties.get(TARGET_BASE_PATH)).map(PropertyBuilder::normalizePath).orElseThrow(InvalidPathException::new);
+        validatePath(normalizedPathSource);
+        validatePath(normalizedPathTarget);
         JobProperties jobProperties = new JobProperties();
-        jobProperties.setSourceBasePath(sparkProperties.get(SOURCE_BASE_PATH));
-        jobProperties.setTargetBasePath(sparkProperties.get(TARGET_BASE_PATH));
+        jobProperties.setSourceBasePath(normalizedPathSource);
+        jobProperties.setTargetBasePath(normalizedPathTarget);
         jobProperties.setSourceDeleteFiles(Boolean.valueOf(sparkProperties.getOrDefault(SOURCE_DELETE_FILES, "False")));
         return jobProperties;
     }
 
+    private static void validatePath(String normalizedPath)
+            throws URISyntaxException, IOException, InvalidPathException {
+        URI uriFile = new URI(normalizedPath);
+        FileSystem fileSystem = getFileSystemForFiles(uriFile);
+        Path filePath = getPathForFileURI(uriFile);
+        if(pathExists(fileSystem, filePath)){
+            if(isNotADirectory(fileSystem, filePath)){
+                throw new InvalidPathException(String.format("Path to JobProperties don't is a directory, normalized path: %s", normalizedPath));
+            }
+        }else{
+            throw new InvalidPathException(String.format("Path to JobProperties don't exists, normalized path: %s", normalizedPath));
+        }
+    }
+
+    private static FileSystem getFileSystemForFiles(URI fileUri) throws IOException {
```

Comment:
```suggestion
    private static FileSystem getFileSystemForFileURI(URI fileUri) throws IOException {
```
sorry nitpicking again

Reply:
Thanks Ricardo, I changed it on this commit: https://github.com/Leaf-Agriculture/billing-data-aggregation-job/pull/6/commits/17a143be1a75a42bd0df49c3041efbed61424414 
and here: https://github.com/Leaf-Agriculture/billing-data-aggregation-job/pull/6/commits/79bd23f2c5264c7c56bfe72f2ba907277ee4c761